kolibri/plugins/coach/assets/src/vue/exercise-detail-view/index.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+
+  <div id="whole-page">
+    <div id="top-bar">
+      <router-link id="back-btn" :to="pageLink">
+        <mat-svg
+          class="back svg-back"
+          category="navigation"
+          name="arrow_back"
+        />
+        <p class="back">
+          <slot name="text"/>
+        </p>
+      </router-link>
+    </div>
+    <div id="page-body">
+      <slot name="body"/>
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+  module.exports = {
+    props: {
+      backPageName: {
+        type: String,
+        required: true,
+      },
+      backPageParams: {
```

Comment:
In general, props that take objects or arrays need additional validation or should be broken up into more props because standard prop `type` info doesn't say enough about the prop's valid values.

This is a somewhat special case since we're describing part of a `vue-router` [location object](https://router.vuejs.org/en/essentials/navigation.html).

In other places in the code, we've used the pattern of passing in the entire link object [like this](https://github.com/learningequality/kolibri/blob/137ccea4f1749f1c1411b9c40e5b26aae556c469/kolibri/plugins/coach/assets/src/vue/reports/all-recent-tabs.vue#L24).

`backPageName` and `backPageParams` could be combined into a single `backPageLink` object prop. This has the advantage that the object structure is documented, and we could eventually write a single location object validator function to use all over the app.